webgui/scripts/codecompass/urlHandler.js

```diff
@@ -71,23 +98,81 @@ function (lang, hash, topic) {
         return state[key];
 
       var result = {};
-      
+
       var h = hash();
-      
+
       h.split('&').forEach(function (element) {
         var pos = element.indexOf('=');
         if (pos !== -1)
           result[element.substr(0, pos)] = element.substr(pos + 1);
       });
-      
+
       return result;
     },
-    
+
     /**
      * This function returns the current URL as string.
      */
     getURL : function () {
       return document.URL;
+    },
+
+    /**
+     * This function returns the thrift FileInfo which associated with the URL
+     * `fid` state.
+     */
+    getFileInfo : function () {
+      return fileInfo;
+    },
+
+    /**
+     * This function returns the file content which associated with the URL 
+     * `fid` state.
+     */
+    getFileContent : function () {
+      return fileContent;
+    },
+
+    /**
+     * This function returns a Thrift service object which associated with the
+     * URL `fid` state.
+     */
+    getService : function () {
+      return service;
+    },
+
+    /**
+     * This function returns the AST node information which associated with the
+     * URL `node` state.
+     */
+    getAstNodeInfo : function () {
+      return astNodeInfo;
+    },
+
+    /**
+     * This function caches database information which associated with the URL
+     * states such as file information, file content, service etc.
+     */
+    _cacheDbInformation : function () {
+      for (var s in state)
+        switch (s) {
+          case 'fid':
+            fileInfo = model.project.getFileInfo(state[s]);
+            if (fileInfo) {
```

Comment:
This `if` is not needed because an exception is thrown when `getFileInfo()` is called with an invalid file ID.